x-pack/plugins/spaces/server/lib/request_interceptors/on_request_interceptor.ts

```diff
@@ -12,14 +12,12 @@ import {
 import { format } from 'url';
 import { DEFAULT_SPACE_ID } from '../../../common/constants';
 import { modifyUrl } from '../utils/url';
-import { LegacyAPI } from '../../plugin';
 import { getSpaceIdFromPath } from '../../../common';
 
 export interface OnRequestInterceptorDeps {
-  getLegacyAPI(): LegacyAPI;
   http: CoreSetup['http'];
 }
-export function initSpacesOnRequestInterceptor({ getLegacyAPI, http }: OnRequestInterceptorDeps) {
+export function initSpacesOnRequestInterceptor({ http }: OnRequestInterceptorDeps) {
```

Comment:
__nit:__ looks like we can remove `getLegacyAPI` from `on_post_auth_interceptor` as well :tada: